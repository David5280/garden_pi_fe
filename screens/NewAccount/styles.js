import { StyleSheet, Dimensions } from 'react-native';

export default StyleSheet.create({
  container: {
    // alignItems: 'center',
    backgroundColor: '#73A686',
    height: Dimensions.get('window').height,
    justifyContent: 'space-evenly',
    width: Dimensions.get('window').width
  },
  infoContainer: {
    alignItems: 'center',
    backgroundColor: 'rgba(115, 166, 134, .7)',
    borderColor: '#A14550',
    borderRadius: 30,
    borderWidth: 2,
    height: Dimensions.get('window').height * .65,
    margin: 5,
    padding: 10,
    shadowColor: '#000',
    shadowOffset: { width: 2, height: 2 },
    shadowOpacity: 1,
    width: Dimensions.get('window').width * .95,
  },
  formContainer: {
    backgroundColor: "rgb(192,192,192)",
    borderColor: '#A14550',
    borderRadius: 30,
    borderWidth: 2,
    flexDirection: 'column',
    height: Dimensions.get('window').height * .50,
    justifyContent: 'space-around',
    alignItems: 'center',
    padding: 25,
  },
  formInput: {
    alignItems: 'center',
    backgroundColor: "#fff",
    borderColor: '#A14550',
    borderRadius: 30,
    borderWidth: 1,
    height: 40,
    justifyContent: 'center',
    paddingLeft: 10,
    width: Dimensions.get('window').width * .60,
    
  },
  errMsgContainer: {
    height: 15
  },
  backBtn: {
    width: 35,
    height: 35
  },  
});

