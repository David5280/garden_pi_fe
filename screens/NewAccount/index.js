import React, { Component } from 'react';
import { 
  View, 
  TouchableOpacity, 
  TextInput,
  Text, 
  ImageBackground, 
  Image } from 'react-native';
import { Header } from '../../components';
import { createNewUser } from '../../Api/ApiCalls';
import styles from './styles';

export default class Splash extends Component {
  state = {
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    passwordConfirmation: '',
    hasErrored: false
  };

  clearState = () => {
    this.setState({
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      passwordConfirmation: '',
    hasErrored: false
    });
  };

  onBackPress = () => {
    this.props.navigation.navigate('Splash')
  };

  onSubmit = () => {
    createNewUser(this.state)
    this.clearState()
  };

  render() {
    return (
      <ImageBackground 
        source={require('../../assets/images/splash-background.jpg')}
        style={styles.container}
      >
        <View style={styles.infoContainer}>
          <View style={{ flexDirection: 'row', alignItems: 'center' }}>
            <TouchableOpacity onPress={this.onBackPress}>
              <Image  
              source={require('../../assets/images/back.png')}
              style={[styles.backBtn, { marginRight: 10 }]}
              />
            </TouchableOpacity>
            <Header style={styles.header} fontsize={35}/>
          </View>
          <View style={styles.formContainer}>
            <TextInput 
              value={this.state.firstName}
              style={styles.formInput}
              placeholder="First Name..."
              maxLength={20}
              onChangeText={(text) => this.setState({ firstName: text })}
            />
            <TextInput           
              value={this.state.lastName}
              style={styles.formInput}
              placeholder="Last Name..."
              maxLength={20}
              onChangeText={(text) => this.setState({ lastName: text })}
            />
            <TextInput 
              value={this.state.email}
              style={styles.formInput}
              placeholder="Email Address..."
              onChangeText={(text) => this.setState({ email: text })}
            />
            <TextInput 
              value={this.state.password}
              style={styles.formInput}
              placeholder="Password..."
              maxLength={20}
              secureTextEntry={true}
              secureTextEntry
              autoCorrect={false}
              onChangeText={(text) => this.setState({ password: text })}
            />
            <TextInput 
              value={this.state.passwordConfirmation}
              style={styles.formInput}
              placeholder="Confirm Password..."
              maxLength={20}
              secureTextEntry
              autoCorrect={false}
              onChangeText={(text) => this.setState({ passwordConfirmation: text })}
            />
            <View style={styles.errMsgContainer}>
              {this.state.hasErrored && <Text>Whoops! Something is Wrong.</Text>}
            </View>
            <TouchableOpacity style={styles.formInput} onPress={this.onSubmit}>
              <Text style={{ fontWeight: 'bold' }}>Create Account!</Text>
            </TouchableOpacity>
          </View>
        </View>
      </ImageBackground>
    )
  }
};