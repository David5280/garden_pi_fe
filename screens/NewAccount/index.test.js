import React from 'react';
import NewAccount from './index';
import 'react-native';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<NewAccount />).toJSON();
  expect(tree).toMatchSnapshot();
});