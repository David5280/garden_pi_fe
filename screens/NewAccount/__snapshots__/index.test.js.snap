// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#73A686",
      "height": 1334,
      "justifyContent": "space-evenly",
      "width": 750,
    }
  }
>
  <Image
    source=""
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source=""
          style={
            Array [
              Object {
                "height": 35,
                "width": 35,
              },
              Object {
                "marginRight": 10,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "lemondrop",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "textShadowRadius": 5,
              },
              Object {
                "color": "#E64C3C",
                "fontSize": 50,
              },
            ]
          }
        >
          G
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "lemondrop",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "textShadowRadius": 5,
              },
              Object {
                "color": "#FF9933",
                "fontSize": 50,
              },
            ]
          }
        >
          A
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "lemondrop",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "textShadowRadius": 5,
              },
              Object {
                "color": "#F1C431",
                "fontSize": 50,
              },
            ]
          }
        >
          R
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "lemondrop",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "textShadowRadius": 5,
              },
              Object {
                "color": "#187F03",
                "fontSize": 50,
              },
            ]
          }
        >
          D
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "lemondrop",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "textShadowRadius": 5,
              },
              Object {
                "color": "#3998DB",
                "fontSize": 50,
              },
            ]
          }
        >
          E
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "lemondrop",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "textShadowRadius": 5,
              },
              Object {
                "color": "#4b0082",
                "fontSize": 50,
              },
            ]
          }
        >
          N
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "lemondrop",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "textShadowRadius": 5,
              },
              Object {
                "color": "#7f00ff",
                "fontSize": 50,
              },
            ]
          }
        >
          Pi
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(115, 166, 134, .9)",
          "borderColor": "#A14550",
          "borderRadius": 30,
          "borderWidth": 2,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 100,
          "marginLeft": 20,
          "padding": 25,
          "width": 562.5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        maxLength={20}
        onChangeText={[Function]}
        placeholder="First Name..."
        rejectResponderTermination={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d5fdd5",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 10,
            "paddingLeft": 10,
            "width": 487.5,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        maxLength={20}
        onChangeText={[Function]}
        placeholder="Last Name..."
        rejectResponderTermination={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d5fdd5",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 10,
            "paddingLeft": 10,
            "width": 487.5,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Email Address..."
        rejectResponderTermination={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d5fdd5",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 10,
            "paddingLeft": 10,
            "width": 487.5,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        maxLength={20}
        onChangeText={[Function]}
        placeholder="Password..."
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d5fdd5",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 10,
            "paddingLeft": 10,
            "width": 487.5,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        maxLength={20}
        onChangeText={[Function]}
        placeholder="Confirm Password..."
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d5fdd5",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 10,
            "paddingLeft": 10,
            "width": 487.5,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 20,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#d5fdd5",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Create Account!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
